numba/tests/test_unicode.py

```diff
@@ -0,0 +1,254 @@
+                                 cfunc(a, b),
+                                 '%s, %s' % (a, b))
+
+    def test_endswith(self, flags=no_pyobj_flags):
+        pyfunc = endswith_usecase
+        cfunc = njit(pyfunc)
+        for a in UNICODE_EXAMPLES:
+            for b in [x for x in ['', 'x', a[:-2], a[3:], a, a + a]]:
+                self.assertEqual(pyfunc(a, b),
+                                 cfunc(a, b),
+                                 '%s, %s' % (a, b))
+
+    def test_in(self, flags=no_pyobj_flags):
+        pyfunc = in_usecase
+        cfunc = njit(pyfunc)
+        for a in UNICODE_EXAMPLES:
+            extras = ['', 'xx', a[::-1], a[:-2], a[3:], a, a + a]
+            for substr in [x for x in extras]:
+                self.assertEqual(pyfunc(substr, a),
+                                 cfunc(substr, a),
+                                 "'%s' in '%s'?" % (substr, a))
+
+    def test_find(self, flags=no_pyobj_flags):
+        pyfunc = find_usecase
+        cfunc = njit(pyfunc)
+        for a in UNICODE_EXAMPLES:
+            extras = ['', 'xx', a[::-1], a[:-2], a[3:], a, a + a]
+            for substr in [x for x in extras]:
+                self.assertEqual(pyfunc(a, substr),
+                                 cfunc(a, substr),
+                                 "'%s'.find('%s')?" % (a, substr))
+
+    def test_getitem(self):
+        pyfunc = getitem_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            for i in range(-len(s)):
+                self.assertEqual(pyfunc(s, i),
+                                 cfunc(s, i),
+                                 "'%s'[%d]?" % (s, i))
+
+    def test_slice2(self):
+        pyfunc = getitem_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            for i in list(range(-len(s), len(s))):
+                for j in list(range(-len(s), len(s))):
+                    sl = slice(i, j)
+                    self.assertEqual(pyfunc(s, sl),
+                                     cfunc(s, sl),
+                                     "'%s'[%d:%d]?" % (s, i, j))
+
+    def test_slice3(self):
+        pyfunc = getitem_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            for i in range(-len(s), len(s)):
+                for j in range(-len(s), len(s)):
+                    for k in [-2, -1, 1, 2]:
+                        sl = slice(i, j, k)
+                        self.assertEqual(pyfunc(s, sl),
+                                         cfunc(s, sl),
+                                         "'%s'[%d:%d:%d]?" % (s, i, j, k))
+
+    def test_concat(self, flags=no_pyobj_flags):
+        pyfunc = concat_usecase
+        cfunc = njit(pyfunc)
+        for a in UNICODE_EXAMPLES:
+            for b in UNICODE_EXAMPLES[::-1]:
+                self.assertEqual(pyfunc(a, b),
+                                 cfunc(a, b),
+                                 "'%s' + '%s'?" % (a, b))
+
```

Comment:
This needs test cases that trigger `IndexError: string index out of range`

Reply:
error tests added